### ISSUE TITLE
Notes throw errors if there are no visible comments

### ISSUE THREAD

**Issue by gravitystorm** [openstreetmap/openstreetmap-website#3450]
It's not possible to view a note if there are no visible comments. This can come about either:

* There's only one comment (the original) and the user is deleted (see #3398)
* There's only one comment (the original) and that comment is hidden (see #3438)

In both cases, you get errors, albeit different ones in each case. But the point remains that there's a lot of (untested) logic that assumes there's at least one note comment visible.

Related:
* #3396 
* https://github.com/gravitystorm/openstreetmap-website/issues/6

**Comment by mmd-osm** [comment]
I thought I tested a scenario where all notes were hidden, and I don't remember seeing an error message for non-moderators. Hmm.

**Comment by mmd-osm** [comment]
Good point. What would be your suggestion here? Should the new API endpoint block any attempts to delete the first comment?

**Comment by gravitystorm** [comment]
> Good point. What would be your suggestion here?

I don't know. I think relying on the fact that the first comment has the "opened" event, and that there's no other "opened" events, seems fragile - but I'm not entirely sure why I feel that. But maybe that's a workaround until we sort things out properly?

**Comment by mmd-osm** [comment]
Hmm. An "opened" event is used exclusively when creating a new note. https://github.com/openstreetmap/openstreetmap-website/blob/master/app/controllers/api/notes_controller.rb#L76

I think what needs to be clarified first is whether we want the opening event to be hidden at all, and other parts of the code need to be adjusted to handle this case properly.

**Comment by gravitystorm** [comment]
I've moved the discussion of the "opened" comment to the PR in #3438 , since this issue is to deal with the wider problems of the current implementation.

**Comment by AntonKhorev** [comment]
Fixed in #5609
